src/MongoDB.Driver.Core/Core/Configuration/ClusterSettings.cs

```diff
@@ -75,7 +85,21 @@ public class ClusterSettings
             Optional<IReadOnlyDictionary<string, BsonDocument>> schemaMap = default(Optional<IReadOnlyDictionary<string, BsonDocument>>),
             Optional<ConnectionStringScheme> scheme = default(Optional<ConnectionStringScheme>))
         {
+            _connectionModeSwitch = connectionModeSwitch.WithDefault(ConnectionModeSwitch.NotSet);
+#pragma warning disable CS0618
+            if (connectionMode.HasValue && !connectionModeSwitch.HasValue)
+            {
+                // the ClusterConnectionModeSwitch value has not been provided
+                throw new InvalidOperationException("ClusterSettings.ConnectionMode must be used only with ClusterConnectionModeSwitch.");
+            }
             _connectionMode = connectionMode.WithDefault(ClusterConnectionMode.Automatic);
+#pragma warning restore CS0618
+            if (directConnection.HasValue && !connectionModeSwitch.HasValue)
+            {
+                // the ClusterConnectionModeSwitch value has not been provided
+                throw new InvalidOperationException("ClusterSettings.DirectConnection must be used only with ClusterConnectionModeSwitch.");
+            }
+            _directConnection = directConnection.WithDefault(null);
```

Comment:
Add (at **END** of method):

    ClusterConnectionModeHelper.EnsureConnectionModeValuesAreValid(_connectionMode, _connectionModeSwitch, _directConnection);